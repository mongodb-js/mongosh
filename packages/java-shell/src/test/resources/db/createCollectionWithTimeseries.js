// before
db.coll.drop();
// command
db.createCollection('coll', {
    timeseries: {
        timeField: "timestamp",
        metaField: "metadata",
        granularity: "minutes"
    },
    expireAfterSeconds: 60 * 60 * 24 * 365 * 10
})
// command
db.coll.insertOne(
    {
        "metadata": { "sensorId": 5578, "type": "temperature" },
        "timestamp": new ISODate("2022-09-09T00:00:00.000Z"),
        "temp": 12
    });
// command
db.coll.findOne({
    "timestamp": new ISODate("2022-09-09T00:00:00.000Z")
}, {_id: 0, timestamp: 1});
// clear
db.coll.drop();