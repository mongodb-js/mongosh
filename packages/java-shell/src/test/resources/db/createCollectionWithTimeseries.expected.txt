{ "acknowledged": true, "insertedId": <ObjectID> }
{ "timestamp": {"$date": {"$numberLong": "1662681600000"}} }